app-editors/atom/atom-1.12.2.ebuild

```diff
@@ -0,0 +1,400 @@
+	https://github.com/atom/git-utils/archive/v${GIT_UTILS_V}.tar.gz -> atom-git-utils-${GIT_UTILS_V}.tar.gz
+	https://github.com/atom/keyboard-layout/archive/v${KEYBOARD_LAYOUT_V}.tar.gz -> atom-keyboard-layout-${KEYBOARD_LAYOUT_V}.tar.gz
+	https://github.com/atom/marker-index/archive/v${MARKER_INDEX_V}.tar.gz -> atom-marker-index-${MARKER_INDEX_V}.tar.gz
+	https://github.com/atom/node-keytar/archive/v${NODE_KEYTAR_V}.tar.gz -> atom-node-keytar-${NODE_KEYTAR_V}.tar.gz
+	https://github.com/atom/node-nslog/archive/v${NODE_NSLOG_V}.tar.gz -> atom-node-nslog-${NODE_NSLOG_V}.tar.gz
+	https://github.com/atom/node-oniguruma/archive/v${NODE_ONIGURUMA_V}.tar.gz -> atom-node-oniguruma-${NODE_ONIGURUMA_V}.tar.gz
+	https://github.com/atom/node-pathwatcher/archive/v${NODE_PATHWATCHER_V}.tar.gz -> atom-node-pathwatcher-${NODE_PATHWATCHER_V}.tar.gz
+	https://github.com/atom/node-runas/archive/v${NODE_RUNAS_V}.tar.gz -> atom-node-runas-${NODE_RUNAS_V}.tar.gz
+	https://github.com/atom/scrollbar-style/archive/v${SCROLLBAR_STYLE_V}.tar.gz -> atom-scrollbar-style-${SCROLLBAR_STYLE_V}.tar.gz
+	https://github.com/atom/node-spellchecker/archive/v${NODE_SPELLCHECKER_V}.tar.gz -> atom-node-spellchecker-${NODE_SPELLCHECKER_V}.tar.gz
+"
+
+BINMODS="
+		cached-run-in-this-context
+		node-ctags
+		git-utils
+		keyboard-layout
+		node-nslog
+		node-oniguruma
+		node-pathwatcher
+		node-runas
+		node-keytar
+		scrollbar-style
+		node-spellchecker
+		marker-index
+"
+
+RESTRICT="mirror"
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+
+DEPEND="
+	${PYTHON_DEPS}
+	>=app-text/hunspell-1.3.3:=
+	>=dev-libs/libgit2-0.23:=[ssh]
+	>=gnome-base/libgnome-keyring-3.12:=
+	>=dev-libs/oniguruma-5.9.5:=
+	>=dev-util/ctags-5.8
+	>=dev-util/electron-1.3.5:${ELECTRON_SLOT}
+"
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/${PN}-${MY_PV}"
+
+pkg_setup() {
+	python-any-r1_pkg_setup
+}
+
+get_install_suffix() {
+	local c=(${SLOT//\// })
+	local slot=${c[0]}
+	local suffix
+
+	if [[ "${slot}" == "0" ]]; then
+		suffix=""
+	else
+		suffix="-${slot}"
+	fi
+
+	echo -n "${suffix}"
+}
+
+get_install_dir() {
+	echo -n "/usr/$(get_libdir)/atom$(get_install_suffix)"
+}
+
+get_electron_dir() {
+	echo -n "/usr/$(get_libdir)/electron-${ELECTRON_SLOT}"
+}
+
+get_electron_nodedir() {
+	echo -n "/usr/include/electron-${ELECTRON_SLOT}/node/"
+}
+
+enode_electron() {
+	"$(get_electron_dir)"/node $@
+}
+
+enodegyp_atom() {
+	local apmpath="/usr/share/atom/resources/app/apm"
+	local nodegyp="${S}/${apmpath}/node_modules/node-gyp/bin/node-gyp.js"
+
+	PATH="$(get_electron_dir):${PATH}" \
+		enode_electron "${nodegyp}" \
+			--nodedir="$(get_electron_nodedir)" $@ || die
+}
+
+easar() {
+	local asar="${WORKDIR}/$(package_dir asar)/node_modules/asar/bin/asar"
+	echo "asar" $@
+	enode_electron "${asar}" $@ || die
+}
+
+package_dir() {
+	local binmod="${1}" binmod_v
+	eval binmod_v=\${$(tr '[:lower:]' '[:upper:]' <<< ${binmod//-/_}_V)}
```

Comment:
Sorry I missed this one. Can we switch to using ${!foo} type indirections instead?